src/gsTensor/gsTensorDomainIterator.h

```diff
@@ -144,13 +144,25 @@ class gsTensorDomainIterator : public gsDomainIterator<T>
         }
         while ( nextCubeVertex(v, l, u) );
     }
-    
+
     const gsVector<T> & lowerCorner() const
     { return lower; }
 
     const gsVector<T> & upperCorner() const
     { return upper; }
 
+    bool isBoundaryElement() const
+    {
+        if ((lower[0]-*meshStart[0]==0)
```

Comment:
It would be better if this worked for any dimension, this would be:
  return (lower-*meshStart == 0 ).any() || (upper-*meshStart == 0 ).any() ;

Reply:
Do you want me to add the changes in gsOverIntegrateRule::mapTo?

Reply:
This one does not work, I think because `lower` is a `gsVector<T>` and `meshStart` is a `gsVector<uiter,D>`.

Reply:
OK then probably this works:

 return (lower == curElement).any() || (upper==curElement).any() ;

Reply:
Neither..
`/home/hverhelst/Documents/gismo98/src/gsTensor/gsTensorDomainIterator.h:154:10:   required from here
/home/hverhelst/Documents/gismo98/src/gsTensor/gsTensorDomainIterator.h:157:42: error: request for member ‘any’ in ‘((const Eigen::MatrixBase<Eigen::Matrix<__gnu_cxx::__normal_iterator<const double*, std::vector<double, std::allocator<double> > >, 4, 1, 0, 4, 1> >*)(&((const gismo::gsTensorDomainIterator<double, 4>*)this)->gismo::gsTensorDomainIterator<double, 4>::meshStart))->Eigen::MatrixBase<Eigen::Matrix<__gnu_cxx::__normal_iterator<const double*, std::vector<double, std::allocator<double> > >, 4, 1, 0, 4, 1> >::operator==<Eigen::Matrix<__gnu_cxx::__normal_iterator<const double*, std::vector<double, std::allocator<double> > >, 4, 1, 0, 4, 1> >(((const gismo::gsTensorDomainIterator<double, 4>*)this)->gismo::gsTensorDomainIterator<double, 4>::curElement)’, which is of non-class type ‘bool’
  157 |         return (meshStart == curElement).any() || (meshEnd==curElement).any() ;
`

Reply:
This works:
`return (meshStart.array() == curElement.array()).any() || ((meshEnd.array())==curElement.array()).any() ;`.

Only the `meshEnd.array()-1` will give errors

Reply:
Another trial: 
return (meshStart.array() == curElement.array()).any() || ((meshEnd.array())-curElement.array()==1).any() ;
or 
return (meshStart.array() == curElement.array()).any() || ( (meshEnd.-curElement).value().array()==1).any();

Reply:
neither worked..